guidance/models/_azureai_studio.py

```diff
@@ -29,8 +28,12 @@ def __init__(
         azureai_studio_endpoint: str,
         azureai_model_deployment: str,
         azureai_studio_key: str,
-        clear_cache: bool,
+        endpoint_type: str,
     ):
+        if endpoint_type not in _ENDPOINT_TYPES:
+            msg = f"endpoint_type {endpoint_type} not valid"
+            raise ValueError(msg)
+        self.endpoint_type = endpoint_type
```

Comment:
Hmm, seems reasonable to ask the user/present an option here as a first cut. I like a default kwarg more than an entirely separate class.

My original thinking was that we'd just check for the presence of role tags in a prompt and use that to determine if we should go Chat or Completion. Even if we prompt the user, we should probably add a role tag check anyway and throw a warning -- if they specify `with system()` `with assistant()` etc. and set the endpoint to completion, there's a good chance it's unintended behavior. Thoughts? We could do this in a future PR perhaps.